editor.planx.uk/src/pages/FlowEditor/components/Sidebar/Search/SearchResultCard/getDisplayDetailsForResult.tsx

```diff
@@ -0,0 +1,277 @@
+  IndexedNode,
+} from "@opensystemslab/planx-core/types";
+import { Calculate } from "@planx/components/Calculate/model";
+import { Confirmation } from "@planx/components/Confirmation/model";
+import { FileUploadAndLabel } from "@planx/components/FileUploadAndLabel/model";
+import { List } from "@planx/components/List/model";
+import { NextSteps } from "@planx/components/NextSteps/model";
+import { ChecklistField } from "@planx/components/shared/Schema/model";
+import { TaskList } from "@planx/components/TaskList/model";
+import { SearchResult } from "hooks/useSearch";
+import { capitalize, get } from "lodash";
+import { SLUGS } from "pages/FlowEditor/data/types";
+import { useStore } from "pages/FlowEditor/lib/store";
+
+/**
+ * Functions to map a search result to the fields required by SearchResultCard
+ */
+interface SearchResultFormatter {
+  getDisplayKey: (result: SearchResult<IndexedNode>) => string;
+  getIconKey: (result: SearchResult<IndexedNode>) => ComponentType;
+  getTitle: (result: SearchResult<IndexedNode>) => string;
+  getHeadline: (result: SearchResult<IndexedNode>) => string;
+  getComponentType: (result: SearchResult<IndexedNode>) => string;
+}
+
+type KeyMap = Record<string, Partial<SearchResultFormatter>>;
+
+type ComponentMap = Partial<
+  Record<ComponentType, Partial<SearchResultFormatter>>
+>;
+
+const keyFormatters: KeyMap = {
+  "data.fn": {
+    getDisplayKey: () => "Data",
+  },
+  "data.val": {
+    getDisplayKey: ({ item }) =>
+      item.type === ComponentType.Answer ? "Option (data)" : "Option",
+  },
+  "data.fileTypes.fn": {
+    getDisplayKey: () => "File type (data)",
+    getHeadline: ({ item, refIndex }) =>
+      (item.data as unknown as FileUploadAndLabel).fileTypes[refIndex].fn!,
+  },
+  "data.dataFieldBoundary": {
+    getDisplayKey: () => "Field boundary",
+  },
+  "data.dataFieldArea": {
+    getDisplayKey: () => "Field area",
+  },
+  "data.description": {
+    getDisplayKey: ({ item }) =>
+      item.type === ComponentType.Answer
+        ? "Option (description)"
+        : "Description",
+  },
+  "data.notes": {
+    getDisplayKey: () => "Internal notes",
+  },
+  "data.howMeasured": {
+    getDisplayKey: () => "How is it defined",
+  },
+  "data.policyRef": {
+    getDisplayKey: () => "Policy reference",
+  },
+  "data.info": {
+    getDisplayKey: () => "Why it matters",
+  },
+  "data.steps.title": {
+    getDisplayKey: () => "Title (step)",
+    getHeadline: ({ item, refIndex }) =>
+      (item.data as unknown as NextSteps).steps[refIndex].title,
+  },
+  "data.steps.description": {
+    getDisplayKey: () => "Description (step)",
+    getHeadline: ({ item, refIndex }) =>
+      (item.data as unknown as NextSteps).steps[refIndex].description,
+  },
+  "data.steps.url": {
+    getDisplayKey: () => "URL (step)",
+    getHeadline: ({ item, refIndex }) =>
+      (item.data as unknown as NextSteps).steps[refIndex].url!,
+  },
+  "data.fileTypes.name": {
+    getDisplayKey: () => "Name",
```

Comment:
nit: I think I'd expect this to be `File type (name)` based on above established pattern for `data.fileTypes.fn` ? Same goes for all `data.fileTypes`-prefixed variables below.

Reply:
Good catch - I'm trying to follow `fieldName (subComponent)` so I've updated these to follow 👍